lib/client.js

```diff
@@ -1718,6 +1728,31 @@ MatrixClient.prototype.loginWithSAML2 = function(relayState, callback) {
     }, callback);
 };
 
+/**
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.getCasServer = function(callback) {
+    return this._http.authedRequest(
```

Comment:
is `authedRequest` really what you want to be calling here? Presumably the user has not logged in yet and so has no credentials? `this._http.request(` may be more accurate (args remain the same).